src/tracks/ui/AffordanceHandle.cpp

```diff
@@ -0,0 +1,105 @@
+#include "AffordanceHandle.h"
+
+#include "../../HitTestResult.h"
+#include "../../ProjectAudioIO.h"
+#include "../../RefreshCode.h"
+#include "../../ViewInfo.h"
+#include "../../SelectionState.h"
+#include "../../ProjectSettings.h"
+#include "../../TrackPanelMouseEvent.h"
+#include "../../WaveClip.h"
+#include "../../ProjectHistory.h"
+#include "../../Track.h"
+#include "../../WaveTrack.h"
+#include "../../../images/Cursors.h"
+
+UIHandlePtr AffordanceHandle::HitAnywhere(std::weak_ptr<AffordanceHandle>& holder, const std::shared_ptr<Track>& pTrack)
+    {
+        auto result = std::make_shared<AffordanceHandle>(pTrack);
+        result = AssignUIHandlePtr(holder, result);
+        return result;
+    }
+
+HitTestPreview AffordanceHandle::HitPreview(const AudacityProject* WXUNUSED(pProject), bool unsafe, bool moving)
+{
+    static auto disabledCursor =
+        MakeCursor(wxCURSOR_NO_ENTRY, DisabledCursorXpm, 16, 16);
+    static auto handOpenCursor =
+        MakeCursor(wxCURSOR_HAND, RearrangeCursorXpm, 16, 16);
+    static auto handClosedCursor =
+        MakeCursor(wxCURSOR_HAND, RearrangingCursorXpm, 16, 16);
+    // i18n-hint Appears on hovering mouse over clip affordance
+    auto message = XO("Click and drag to move a clip in time, or click to select");
```

Comment:
Design specified this string?

Reply:
No, but we definitely will need another hit text for this handle, do i need to talk to designers about that or strings will be reviewed and updated later before we start next release?

Reply:
I made a card for this question in the project board.